### ISSUE TITLE
[IMPROVEMENT] Improve performance 

### ISSUE THREAD

**Issue by rasputinea** [Esophose/PlayerParticles#28]
Playerparticles is very heavy on my server when a lot of players with particles are on, this could easily be solved by spawning the particles asynchronously. I am slightly confused as to why you don't do this already.

**Comment by Esophose** [comment]
Spawning particles using the Spigot API, to my knowledge, isn't thread-safe so I can't do it asynchronously. If you're using Spigot, can you try using Paper instead, and if you're already using Paper, could you get me a timings report that has recorded for at least 15 minutes so I can try to see if there's a problem somewhere? (Make sure the timings report is taken when you are experiencing server load from the particles)

I have personally spawned 5000+ particles per tick with no issues with server performance. Even other particle plugins such as Block Particles use synchronous particle spawning 🤷‍♀

**Comment by rasputinea** [comment]
https://pastebin.com/xHUuBkb7 this code is thread safe, (p.spawnparticle and iterating through onlineplayers is thread safe also)

**Comment by Esophose** [comment]
See ddbfd5292ed660e6bc44c9dd38d9bb7de7139b76

I tried to make as much of the plugin async as I could. There may still have been some things that I've missed, but it's working good as far as I can tell. I'll attach a jar of v7.2 with that commit applied. Once I've verified everything is working perfectly as normal I'll go ahead and release it as a v7.3 update. 

[playerparticles-7.2-async.zip](https://github.com/Esophose/PlayerParticles/files/4209650/playerparticles-7.2-async.zip)
